source/dotnet/Library/AdaptiveCards/ParseContext.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace AdaptiveCards
+{
+    static class ParseContext
+    {
+        public static void Initialize()
+        {
+            ElementIDs = new Dictionary<string, List<AdaptiveInternalID>>();
+            IDStack = new Stack<Tuple<string, AdaptiveInternalID, bool>>();
+        }
+
+        public enum ContextType { Element, Action };
+
+        public static ContextType Type { get; set; }
+
+        private static IDictionary<string, List<AdaptiveInternalID>> ElementIDs = new Dictionary<string, List<AdaptiveInternalID>>();
+
+        private static Stack<Tuple<string, AdaptiveInternalID, bool>> IDStack = new Stack<Tuple<string, AdaptiveInternalID, bool>>();
+
+        // Push the provided state on to our ID stack
+        public static void PushElement(string idJsonProperty, AdaptiveInternalID internalId)
+        {
+            if (internalId.Equals(AdaptiveInternalID.Invalid))
+            {
+                throw new AdaptiveSerializationException($"Attemping to push an element on to the stack with an invalid ID");
+            }
+            IDStack.Push(new Tuple<string, AdaptiveInternalID, bool>(idJsonProperty, internalId, AdaptiveFallbackConverter.IsInFallback));
+        }
+
+        // Pop the last id off our stack and perform validation 
+        public static void PopElement()
+        {
+            // about to pop an element off the stack. perform collision list maintenance and detection.
+            var idsToPop = IDStack.Peek();
+            string elementID = idsToPop.Item1;
+            AdaptiveInternalID elementInternalID = idsToPop.Item2;
+            bool isFallback = idsToPop.Item3;
+
+            if (!string.IsNullOrEmpty(elementID))
```

Comment:
>IsNullOrEmpty [](start = 24, length = 13)

IsNullOrWhitespace

Reply:
see previous comment on the subject :)

---
In reply to: [278345361](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278345361) [](ancestors = 278345361)